index.js

```diff
@@ -2,268 +2,299 @@
 
 'use strict';
 
-const tls = require('tls');
-const net = require('net');
-const eos = require('end-of-stream');
-const through = require('through2');
-const minimist = require('minimist');
 const allContainers = require('docker-allcontainers');
-const statsFactory = require('docker-stats');
-const logFactory = require('docker-loghose');
+const eos = require('end-of-stream');
 const eventsFactory = require('docker-event-log');
+const logFactory = require('docker-loghose');
+const net = require('net');
 const os = require('os');
+const statsFactory = require('docker-stats');
+const through = require('through2');
+const tls = require('tls');
+const winston = require('winston');
 
-let debugLogging = !!process.env.INSIGHT_DOCKER_DEBUG;
+const { Command } = require('commander');
 
-const logDebug = (...args) => {
-  if (!debugLogging) {
-    return;
-  }
 
-  console.log(...args);
-};
+//  Winston logger initialised after CLI arg parsing
+var LOGGER;
+
 
 function connect(opts) {
   let stream;
   const endpoint = `${opts.region}${opts.server}`;
+
   if (opts.secure) {
-    logDebug(`Establishing secure connection to ${endpoint}:${opts.port}...`);
+    LOGGER.info(`Establishing secure connection to ${endpoint}:${opts.port}`);
     stream = tls.connect(opts.port, endpoint, onSecure);
   } else {
-    logDebug(`Establishing plain-text connection to ${endpoint}:${opts.port}...`);
+    LOGGER.info(`Establishing plain-text connection to ${endpoint}:${opts.port}`);
     stream = net.createConnection(opts.port, endpoint);
   }
 
   function onSecure() {
-    // let's just crash if we are not secure
     if (!stream.authorized) {
-      logDebug('Connection is not secure!');
-      throw new Error('secure connection not authorized');
+      // let's just crash if we are not secure
+      throw new Error('Secure connection is not authorized');
     }
+    LOGGER.info('Connection is secure');
   }
 
   return stream;
 }
 
 
 function start(opts) {
-  debugLogging = opts.debug;
-  const logsToken = opts.logstoken || opts.token;
-  const statsToken = opts.statstoken || opts.token;
-  const eventsToken = opts.eventstoken || opts.token;
   let out;
   let noRestart = () => void 0;
 
   const filter = through.obj(function (obj, enc, cb) {
-    logDebug(`Got an event with encoding "${enc}":`, obj);
+    LOGGER.debug(`Got an event with encoding "${enc}"`, obj);
```

Comment:
```suggestion
    LOGGER.debug(`Got an event with encoding "${enc}":`, obj);
```
we're printing the received event after this